types/auth0/index.d.ts

```diff
@@ -1112,9 +1112,77 @@ export class AuthenticationClient {
     ): void;
 }
 
+export interface ActionsManagerOptions {
+    baseUrl: string;
+    headers?: any;
+    retry?: RetryOptions;
+}
+
+export class ActionsManager {
+    constructor(options: ActionsManagerOptions);
+
+    create(data: object, cb: (err: Error) => void): void;
+    create(data: object): Promise<void>;
+
+    createVersion(params: ObjectWithId, data: object, cb: (err: Error, actionVersion: string) => void): void;
```

Comment:
@typescript/eslint warns on uses of `object`: "Don't use `object` as a type. The `object` type is currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)).
Consider using `Record<string, unknown>` instead, as it allows you to more easily inspect and use the keys"

The actions api does give a good amount of information on the body expected for these create calls. Did you have a justification for using `object` instead of a specific parameters type of even `any` as is used in other parts of the types?

Reply:
> The actions api does give a good amount of information on the body expected for these create calls.

Do you mean this API documentation? https://auth0.com/docs/api/management/v2#!/Actions/get_actions

Or is there any other document that more concretely specifies the exact schemas and not just examples of it? I guess I could update the branch to declare more specific types based on the response examples for each endpoint, but I'm worried it still wouldn't be 100% exact, thinking about enum types represented as strings for which the possible values are not specified in the docs, etc.

Reply:
Yes, that's the documentation I was referring to. You're right that it's not a strict schema, and you'd have to infer some things based on the example. My basic issue is the use of the `object` type rather than a specific one or `any`, if we can't get specific schema info. I don't like using `any` but it is used in other similar places in this types package.

Reply:
Fair point. I'll go ahead and make the `ActionsManager` types a bit more specific according to the documentation where they can be inferred safely. Thanks for the suggestion.

Reply:
Eh, this has started to take a lot longer than I previously expected. There are some discrepancies between the management API docs and the node-auth0 lib docs.

| Endpoint | Management API docs | node-auth0 docs |
| ------------- | ------------- | ------------- |
| Create an action | Responds with the action created. [(link)](https://auth0.com/docs/api/management/v2/#!/Actions/post_action) | Doesn't expect the action to be returned. [(link)](https://auth0.github.io/node-auth0/module-management.ActionsManager.html#create) |
| Create a version | Undocumented. | [(link)](https://auth0.github.io/node-auth0/module-management.ActionsManager.html#createVersion) |
| Get all triggers | | Code example makes the impression that it returns actions. [(link)](https://auth0.github.io/node-auth0/module-management.ActionsManager.html#getAllTriggers) |
| Get trigger bindings | `triggerId` request param is camelcase, endpoint accepts paging options. [(link)](https://auth0.com/docs/api/management/v2/#!/Actions/get_bindings) | `trigger_id` is snake_case, doesn't accept paging options. [(link)](https://auth0.github.io/node-auth0/module-management.ActionsManager.html#getTriggerBindings) |
| Get a specific version of an action | Uses `actionId` for action ID and `id` for version ID. [(link)](https://auth0.com/docs/api/management/v2/#!/Actions/get_action_version) | Uses `version_id` for version ID and `id` for action ID. [(link)](https://auth0.github.io/node-auth0/module-management.ActionsManager.html#getVersion) |
| Get an action's versions | Uses `actionId`. [(link)](https://auth0.com/docs/api/management/v2/#!/Actions/get_action_versions) | Uses `id`. [(link)](https://auth0.github.io/node-auth0/module-management.ActionsManager.html#getVersions) |
| Test an action | Returns payload in response. [(link)](https://auth0.com/docs/api/management/v2/#!/Actions/post_test_action) | Doesn't seem to expect a response body. [(link)](https://auth0.github.io/node-auth0/module-management.ActionsManager.html#test) |
| Test a trigger | Undocumented. | [(link)](https://auth0.github.io/node-auth0/module-management.ActionsManager.html#testTrigger) |
| Update trigger bindings | Uses `triggerId`. [(link)](https://auth0.com/docs/api/management/v2/#!/Actions/patch_bindings) | Uses `trigger_id`, and the example shows a payload that doesn't remotely match the management API docs. [(link)](https://auth0.github.io/node-auth0/module-management.ActionsManager.html#updateTriggerBindings) |

Considering the cumbersome and manual nature of having to untangle these ambiguities one by one I don't think I'll be able to proceed with this pull request.